test/LaunchDarkly.EventSource.Tests/BaseTest.cs

```diff
@@ -39,6 +40,14 @@ public BaseTest()
             _logCapture = Logs.Capture();
             _testLogging = _logCapture;
             _testLogger = _testLogging.Logger("");
+
+            // The following line prevents intermittent test failures that can happen due to the low
+            // default setting of ThreadPool.SetMinThreads causing new worker tasks to be severely
+            // throttled: http://joeduffyblog.com/2006/07/08/clr-thread-pool-injection-stuttering-problems/
+            // This makes it difficult to test things such as timeouts. We believe it not to be a real
+            // issue in non-test scenarios, since the tests are starting and stopping an unusually large
+            // number of async tasks in a way that regular use of EventSource would not do.
+            ThreadPool.SetMinThreads(100, 100);
```

Comment:
This had bitten me before when I was debugging the `dotnet-server-sdk` tests and seeing that a lot of tests either took a couple seconds just to start up, or had an inexplicable delay of a couple seconds prior to some trivial async operation (you can see this locally if you just remove this line and run the tests: even if you don't get spurious failures for timeout-related stuff, you'll see that they run much, much slower).

What I found was that, at least on some runtimes, the thread pool size in the Xunit environment is _two_. And if at any time you try to start a Task of any kind (even if it's just a `Task.Delay` that doesn't in itself _do_ anything) when there are already two threads in use from the pool, the scheduler can decide to throttle you by simply pausing for a couple seconds, even though not much else is going on.

That's something that could theoretically be an issue in a real application too, depending on how things are configured, but that wouldn't only affect EventSource. I don't think we are creating a particularly large number of tasks in this code compared to any other async library, it's just that we're running dozens of tests within a short period of time, most of which start a new async EventSource client which might not completely finish shutting down before the next test runs.